.ci/e2eTestingMacosDaily.groovy

```diff
@@ -0,0 +1,126 @@
+#!/usr/bin/env groovy
+
+@Library('apm@current') _
+
+pipeline {
+  agent { label 'darwin && orka && x86_64' }
+  environment {
+    REPO = 'e2e-testing'
+    BASE_DIR = "src/github.com/elastic/${env.REPO}"
+    ELASTIC_APM_ACTIVE = "true"
+    ELASTIC_APM_ENVIRONMENT = "ci"
+    ELASTIC_APM_LOG_FILE = "stderr"
+    ELASTIC_APM_LOG_LEVEL = "debug"
+    NOTIFY_TO = credentials('notify-to')
+    PROVIDER = 'remote'
+    BEAT_VERSION = "${params.BEAT_VERSION.trim()}"
+    ELASTIC_AGENT_VERSION = "${params.ELASTIC_AGENT_VERSION.trim()}"
+    ELASTIC_STACK_VERSION = "${params.ELASTIC_STACK_VERSION.trim()}"
+    CLUSTER_NAME = "e2e-testing-${BUILD_ID}-${BRANCH_NAME}-${ELASTIC_STACK_VERSION.replaceAll('\\.', '-')}"
+    LOG_LEVEL = "${params.LOG_LEVEL}"
+    GO111MODULE = 'on'
+  }
+  options {
+    timeout(time: 120, unit: 'MINUTES')
+    buildDiscarder(logRotator(numToKeepStr: '200', artifactNumToKeepStr: '30', daysToKeepStr: '30'))
+    timestamps()
+    ansiColor('xterm')
+    disableResume()
+    durabilityHint('PERFORMANCE_OPTIMIZED')
+    rateLimitBuilds(throttle: [count: 60, durationName: 'hour', userBoost: true])
+    quietPeriod(10)
+  }
+  parameters {
+    string(name: 'ELASTIC_AGENT_VERSION', defaultValue: '8.2.4-SNAPSHOT', description: 'SemVer version of the Elastic Agent to be used for the tests. You can use here the tag of your PR to test your changes')
+    string(name: 'ELASTIC_STACK_VERSION', defaultValue: '8.2.4-SNAPSHOT', description: 'SemVer version of the stack to be used for the tests.')
```

Comment:
If it's working for 8.2, then wdyt about sending this PR against the 8.2 branch?

Reply:
I wonder if we should we keep it for the main branch, as the target stack is in the 8.2 family. In the project we are syncing versions and branches, so it would be a bit weird having main (8.4 today) running the stack in 8.2

Not a blocker though, as the value provided by this PR is huge

Reply:
I was tempted to change it, but since I added the [`auto-bump`](https://github.com/elastic/e2e-testing/pull/2626/files#diff-a868bcd18bb08e52f2ebfb99dd6cccf60d65165712fdc7859a8845a0d2aba374), I'd like to keep it and then validate if the autobump automation works as expected once this PR is merged